runtime/near-vm-runner/src/wasmer_runner.rs

```diff
@@ -0,0 +1,255 @@
+            // (see get_wasm_interrupt_signal_mem()) is accessed.
+            // This address is passed via InternalCtx.interrupt_signal_mem
+            // to the runtime, and is triggered only from do_optimize().
+            // do_optimize() is only called if backend is mentioned in
+            // Run.optimized_backends option, and we don't.
+            RuntimeError::InstanceImage(_) => {
+                panic!("Support instance image errors properly");
+            }
+            RuntimeError::User(data) => {
+                if let Some(err) = data.downcast_ref::<VMLogicError>() {
+                    match err {
+                        VMLogicError::HostError(h) => {
+                            VMError::FunctionCallError(FunctionCallError::HostError(h.clone()))
+                        }
+                        VMLogicError::ExternalError(s) => VMError::ExternalError(s.clone()),
+                        VMLogicError::InconsistentStateError(e) => {
+                            VMError::InconsistentStateError(e.clone())
+                        }
+                    }
+                } else {
+                    panic!(
+                        "Bad error case! Output is non-deterministic {:?} {:?}",
+                        data.type_id(),
+                        self.to_string()
+                    );
+                }
+            }
+        }
+    }
+}
+
+#[cfg(feature = "no_cache")]
+pub(crate) fn compile_module(
+    _code_hash: Vec<u8>,
+    code: &[u8],
+    config: &VMConfig,
+) -> Result<wasmer_runtime::Module, VMError> {
+    let prepared_code = prepare::prepare_contract(code, config)?;
+    wasmer_runtime::compile(&prepared_code).map_err(|err| err.into_vm_error())
+}
+
+pub fn run_wasmer<'a>(
+    code_hash: Vec<u8>,
+    code: &[u8],
+    method_name: &[u8],
+    ext: &mut dyn External,
+    context: VMContext,
+    wasm_config: &'a VMConfig,
+    fees_config: &'a RuntimeFeesConfig,
+    promise_results: &'a [PromiseResult],
+) -> (Option<VMOutcome>, Option<VMError>) {
+    if !cfg!(target_arch = "x86") && !cfg!(target_arch = "x86_64") {
+        // TODO(#1940): Remove once NaN is standardized by the VM.
+        panic!(
+            "Execution of smart contracts is only supported for x86 and x86_64 CPU architectures."
+        );
+    }
+    if method_name.is_empty() {
+        return (
+            None,
+            Some(VMError::FunctionCallError(FunctionCallError::MethodResolveError(
+                MethodResolveError::MethodEmptyName,
+            ))),
+        );
+    }
+
+    let module = match cache::compile_module(code_hash, code, wasm_config) {
+        Ok(x) => x,
+        Err(err) => return (None, Some(err)),
+    };
+    let mut memory = match WasmerMemory::new(
+        wasm_config.limit_config.initial_memory_pages,
+        wasm_config.limit_config.max_memory_pages,
+    ) {
+        Ok(x) => x,
+        Err(_err) => panic!("Cannot create memory for a contract call"),
+    };
```

Comment:
You can replace this `match` with `.expect()` method.

Reply:
Sure, it was just moving existing code, although.